t/t2040-checkout-symlink-attr.sh

```diff
@@ -0,0 +1,54 @@
+#!/bin/sh
+
+test_description='checkout symlinks with `symlink` attribute on Windows
+
+Ensures that Git for Windows creates symlinks of the right type,
+as specified by the `symlink` attribute in `.gitattributes`.'
+
+# Tell MSYS to create native symlinks. Without this flag test-lib's
+# prerequisite detection for SYMLINKS doesn't detect the right thing.
+MSYS=winsymlinks:nativestrict && export MSYS
```

Comment:
It feels icky to set this before testing for the MINGW prerequisite, but I understand why this is necessary.